src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/CounterMetadata.cs

```diff
@@ -6,17 +6,40 @@ namespace Microsoft.Diagnostics.Monitoring.EventPipe
     public class CounterMetadata
     {
         public CounterMetadata(string providerName, string counterName, string meterTags, string instrumentTags, string scopeHash)
+            : this(
+                  providerName,
+                  providerVersion: null,
+                  counterName,
+                  counterUnit: null,
+                  counterDescription: null,
+                  instrumentId: null,
+                  meterTags,
+                  instrumentTags,
+                  scopeHash)
+        {
+        }
+
+        public CounterMetadata(string providerName, string providerVersion, string counterName, string counterUnit, string counterDescription, int? instrumentId, string meterTags, string instrumentTags, string scopeHash)
         {
             ProviderName = providerName;
+            ProviderVersion = providerVersion;
             CounterName = counterName;
+            CounterUnit = counterUnit;
```

Comment:
We already have Unit as part of CounterPayload. You can make that a computed property, but we should not have two units.

Reply:
Updated so the metadata version is used everywhere.